doc: performance considerations for large scale access scopes (#87961)
openshift/openshift-docs: modules/create-a-custom-access-scope.adoc

====
. If you want to allow access to clusters and namespaces based on labels, click *Add label selector* under the *Label selection rules* section. Then click *Add rule* to specify *Key* and *Value* pairs for the label selector. You can specify labels for clusters and namespaces.
. Click *Save*.

> 

Reviewer: I would remove line 49-53 and make "Scale considerations" into a heading like this:

```
[id="scale-considerations_{context}"]
== Scale considerations
```

[NOTE]
====
Scale considerations:

> In large environments (thousands of namespaces), the application of user scopes that cover large numbers of namespaces can lead to some pages being slow to render or even timing out (for example, timeouts when opening the main *Dashboard* page).

Reviewer: ```suggestion
In large environments, for example, with thousands of namespaces, the application of user scopes that cover large numbers of namespaces can cause some pages to render slowly or time out. For example, you might experience timeouts when opening the main *Dashboard* page.
```

> The performances were tested in single-cluster context on an environment with 20000 deployents across 5000 namespaces and 80000 alerts in the one test cluster. In that case, the load time of the main *Dashboard* page for scopes including up to 2000 namespaces on that one cluster was found to be still under a few seconds.

Reviewer: ```suggestion
Performance was tested in a single-cluster context on an environment with 20,000 deployments across 5,000 namespaces and 80,000 alerts in one test cluster. In that testing, the load time of the main *Dashboard* page for scopes including up to 2,000 namespaces on that one cluster was under a few seconds.
```

> The performances were tested in multi-cluster context on an environment with 200000 deployments across 50000 namespaces and 800000 alerts across 10 test clusters. In that case, the load time of the main *Dashboard* page was found to be still under a few seconds for scopes including up to 500 namespaces across the ten clusters as well as for scopes including up to 2000 namespaces on a single cluster.

Reviewer: ```suggestion
Performance was tested in a multi-cluster context on an environment with 200,000 deployments across 50,000 namespaces and 800,000 alerts across 10 test clusters. In that testing, the load time of the main *Dashboard* page was under a few seconds for scopes including up to 500 namespaces across the 10 clusters, as well as for scopes including up to 2,000 namespaces on a single cluster.
```

> The load time of UI pages with user scopes larger than the tested ones cannot be guaranteed to stay under the timeout limit.

Reviewer: ```suggestion
The load time of {product-title-short} portal pages with a larger number of user scopes than the number tested cannot be guaranteed to stay under the timeout limit.
```
